### PR TITLE
Add autonomousapps dependency-analysis-gradle-plugin plugin

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Generates a dependency report for project and library dependencies. This can help us determine what is not used.

Example report output:
https://gist.github.com/JLLeitschuh/53a693d0101576497f5f485b019f8ea8

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
@bot-gradle test RFM

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -126,6 +129,7 @@
          </trusted-key>
          <trusted-key id="d9aa7402ba75f78720d975211d185615d0a84648" group="net.sf.saxon" name="Saxon-HE"/>
          <trusted-key id="df49fa14952e59a7b21931ec8edf2667d0ecffaf" group="io.mockk"/>
+         <trusted-key id="e77417ac194160a3fabd04969a259c7ee636c5ed" group="com.google.errorprone" name="error_prone_annotations"/>
```
Comment (ljacomet) [inline gradle/verification-metadata.xml:132]: Could you make sure all added keys are added to the keyring?

**Review by ljacomet — CHANGES_REQUESTED** [review]
Looks good overall, only comment is about adding keys to the keyring.

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (JLLeitschuh) [reply gradle/verification-metadata.xml:132]: Will do!

**Comment by JLLeitschuh** [comment]
@bot-gradle test QF

**Comment by JLLeitschuh** [comment]
@bot-gradle test this please

**Pull request merged**